WooCommerce/Classes/POS/Models/PointOfSaleAggregateModel.swift

```diff
@@ -201,14 +199,22 @@ extension PointOfSaleAggregateModel {
         }
     }
 
-    func startCashPayment() {
-        isCashPaymentInProgress = true
+    // Prevents card payments from the moment the merchant decides we start collecting cash
+    // Once we get the callback from the card service, we switch to cash collection state
+    @MainActor
+    func startCashPayment() async {
+        try? await cardPresentPaymentService.cancelPayment()
         paymentState = .cash(.collectingCash)
     }
 
-    func cancelCashPayment() {
-        isCashPaymentInProgress = false
-        paymentState = .card(.idle)
+    @MainActor
+    func cancelCashPayment() async {
+        if case .connected = cardReaderConnectionStatus {
+            try? await cardPresentPaymentService.cancelPayment()
+            await collectPayment()
+        } else {
+            paymentState = .card(.idle)
```

Comment:
Perhaps `startPaymentWhenCardReaderConnected` might be a better alternative? Not sure, it works fine as is but you could experiment in a new issue.

Reply:
Logged https://github.com/woocommerce/woocommerce-ios/issues/14859